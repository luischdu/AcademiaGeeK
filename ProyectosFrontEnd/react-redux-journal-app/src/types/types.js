export const types = {
  login: "[Auth] Login",
  logout: "[Auth Logout]",
  uiSetError: "[ui] Set Error",
  uiRemoveError: "[ui] Remuve Error",
  uiStarLogin: "[ui] Start Login",
  uiFinishLogin: "[ui] Finish Login",

  notesAddNew: "[Notes] New notes",
  notesActive: "[Notes] Set notes active",
  notesLoad: "[Notes] Load notes",
  notesUpdate: "[Notes] Update notes",
  notesDelete: "[Notes] Delete notes",

  notesFileUrl: "[Notes] Img url",
  notesLogoutClean: "[Notes]  Logout",
};
