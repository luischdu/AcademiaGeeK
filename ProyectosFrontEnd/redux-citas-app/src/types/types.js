export const types = {
  agregar: "AGREGAR_CITA",
  borrar: "BORRAR_CITA",
  validar: "VALIDAR_CITA",
};
