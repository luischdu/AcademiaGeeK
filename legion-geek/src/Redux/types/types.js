export const types = {
    login: "[Auth] Login",
    logout: "[Auth] Logout",
    recoverPassword: "[Auth] recover password",

    uiSetError: "[ui] Set Error",
    uiRemoveError: "[ui] Remuve Error",
    uiStarLogin: "[ui] Start Login",
    uiFinishLogin: "[ui] Finish Login",
    uiStopLogin: "[ui] Stop Login",

    articlesLoad: "[Articles] Load Articles",
    articlesRead: "[Articles] Read Articles",
    

  };