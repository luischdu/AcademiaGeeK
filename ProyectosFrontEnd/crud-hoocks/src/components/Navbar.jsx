import React from "react";
import { Link, NavLink } from "react-router-dom";

const Navbar = () => {
  return (
    <div>
      <nav className="navbar navbar-expand-lg navbar-light bg-light">
        <div className="container-fluid">
          <Link to="/" className="navbar-brand">
            PLord
          </Link>
          <button
            className="navbar-toggler"
            type="button"
            data-bs-toggle="collapse"
            data-bs-target="#navbarNavAltMarkup"
            aria-controls="navbarNavAltMarkup"
            aria-expanded="false"
            aria-label="Toggle navigation"
          >
            <span className="navbar-toggler-icon"></span>
          </button>
          <div className="collapse navbar-collapse" id="navbarNavAltMarkup">
            <div className="navbar-nav">
              <NavLink
                className="nav-link active"
                aria-current="page"
                exact
                to="/"
              >
                Inicio
              </NavLink>
              <NavLink
                className="nav-link active"
                aria-current="page"
                exact
                to="/add"
              >
                Agregar
              </NavLink>
              <NavLink
                className="nav-link active"
                aria-current="page"
                exact
                to="/edit"
              >
                Editar
              </NavLink>
            </div>
          </div>
        </div>
      </nav>
    </div>
  );
};

export default Navbar;
