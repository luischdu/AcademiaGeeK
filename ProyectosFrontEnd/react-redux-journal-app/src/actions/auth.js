import { types } from "../types/types";
import { firebase, googleAuthProvider } from "../firebase/firebase-config";
import { StartLoading, FinishLoading } from "./uiError";
import Swal from 'sweetalert2'


export const login = (uid, displayName) => {
  return {
    type: types.login,
    payload: {
      uid,
      displayName,
    },
  };
};

export const startLoginEmailPassword = (email, password) => {
  return (dispatch) => {
    dispatch(StartLoading());
    return firebase
      .auth()
      .signInWithEmailAndPassword(email, password)
      .then(({ user }) => {
        dispatch(login(user.uid, user.displayName));
        dispatch(FinishLoading());
      })
      .catch((e) => {
        Swal.fire({
          position: 'center',
          icon: 'error',
          title: `${e.message}`,
          showConfirmButton: false,
          timer: 1500
        })        
      });
  };
};

export const startRegisterWithEmailPasswordName = (email, password, name) => {
  return (dispatch) => {
    firebase
      .auth()
      .createUserWithEmailAndPassword(email, password)
      .then(async ({ user }) => {
        await user.updateProfile({ displayName: name });

        dispatch(login(user.uid, user.displayName));
        console.log(user);
      })
      .catch((e) => {
        Swal.fire({
          position: 'center',
          icon: 'error',
          title: `${e.message}`,
          showConfirmButton: false,
          timer: 1500
        })        
      });
  };
};

export const startGoogleLogin = () => {
  return (dispatch) => {
    firebase
      .auth()
      .signInWithPopup(googleAuthProvider)
      .then(({ user }) => {
        dispatch(login(user.uid, user.displayName));
        console.log(user);
      })
      .catch((e) => {
        Swal.fire({
          position: 'center',
          icon: 'error',
          title: `${e.message}`,
          showConfirmButton: false,
          timer: 1500
        }) 
      });
  };
};

export const logout = () => ({
  type: types.logout
});

export const startLogout = () => {
  return (dispatch) => {
    firebase.auth().signOut();
    dispatch(logout());
  };
};
