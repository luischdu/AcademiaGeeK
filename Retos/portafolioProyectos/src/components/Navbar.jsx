import React, { Component } from 'react'
import 'bootstrap/dist/css/bootstrap.css';
import styled from 'styled-components'
import { faCode } from '@fortawesome/free-solid-svg-icons'
import { Nav, Button, Navbar } from 'react-bootstrap'
import { BodyRegular2 } from '../styles/Body'

const ButtonStyle = styled.nav`
    margin-left: 40%;
    margin-right: 72px;
    @media (max-width: 768px) {
        margin-left: 0;
        margin-right: 0;
        display:none;
    } 
`
const NavbarStyle = styled(Navbar)`
    height: 96px;
    width:100%;
    background-color: #0F0E17;
    padding: 31px 72px 24px 72px;
    position: fixed;
    z-index: 100;
    @media (max-width: 768px) {
        padding:10px;
    }
`


const StyleButton = styled(Button)`
    width: 206px;
    height: 48px;
    border: solid 1.5px #33ccff;
    color: #33ccff;
    &:hover ${StyleButton} {
    background-color: #33ccff;
    border: solid 1.5px #33ccff;
    }
`
const StyledNavLink = styled(Nav.Link)`
    &:hover {
    color: #33ccff;
  }
`


const StyledNavCollapse = styled(Navbar.Collapse)`
    @media (max-width: 768px) {
        background-color: #0F0E17 !important;
        padding: 15px;
        margin-left: -10px;
        margin-right: -10px;
    }
`

const StyledNavBarHeart = styled(Navbar.Brand)`
    @media (max-width: 768px) {
        margin-left: 20%;
    }
`
const StyleButtonCollapse = styled(Button)`
    width: 206px;
    height: 48px;
    display:none;
    border: solid 1.5px #33ccff;
        color: #33ccff;
    &:hover ${StyleButtonCollapse} {
    background-color: #33ccff;
    }
    @media (max-width: 768px) {
        display: block;
        border: solid 1.5px #33ccff;
        color: #33ccff;
    }
`

class NavH extends Component {

    fbShare() {
        window.open('HV_LuisMiguelChávezDuque.pdf');
    }

    render() {
        return (
            <NavbarStyle collapseOnSelect expand="lg" variant="dark" id="navbar">
                <StyledNavBarHeart className="animate__animated animate__heartBeat animate__slower" href="#welcome-section">

                </StyledNavBarHeart>
                <Navbar.Brand>
                    <a href="HV_LuisMiguelChávezDuque.pdf" download="" style={{ textDecoration: "none" }}>
                        <StyleButtonCollapse variant="outline-danger">Descargar Curriculum</StyleButtonCollapse>
                    </a>
                </Navbar.Brand>
                <Navbar.Toggle aria-controls="responsive-navbar-nav" style={{ position: "absolute" }} />
                <StyledNavCollapse id="responsive-navbar-nav">
                    <Nav className="mr-auto">
                        <StyledNavLink href="#welcome"><BodyRegular2 name="Hola" /></StyledNavLink>
                        <StyledNavLink href="#Projects"><BodyRegular2 name="Proyectos" /></StyledNavLink>
                        <StyledNavLink href="#Service"><BodyRegular2 name="Servicios" /></StyledNavLink>
                        <StyledNavLink href="#Testimonials"><BodyRegular2 name="Testimonios" /></StyledNavLink>
                        <StyledNavLink href="#Contact"><BodyRegular2 name="Contacto" /></StyledNavLink>
                    </Nav>
                    <ButtonStyle>
                        <a href="HV_LuisMiguelChávezDuque.pdf" download="" style={{ textDecoration: "none" }}>
                            <StyleButton variant="outline-danger">Descargar Curriculum</StyleButton>
                        </a>
                    </ButtonStyle>
                </StyledNavCollapse>
            </NavbarStyle>
        )
    }
}


export default NavH