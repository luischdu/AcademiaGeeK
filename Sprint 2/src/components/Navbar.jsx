import React from 'react';
import styled, { ThemeProvider } from 'styled-components';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { withRouter } from 'react-router-dom';
import { faUser, faShoppingCart } from '@fortawesome/free-solid-svg-icons';

const HeaderStyle = styled.header`
  display: flex;
  justify-content: space-between;
  align-items: center;
  margin: 9px;
`;

const ImgLogoStyle = styled.img`
  width: 26.5%;
`;

const DivEnd = styled.div`
  display: flex;
  justify-content: flex-end;
  align-content: center;
  gap: 6px;
`;

const Navbar = (props) => {
  
  const { history } = props;

  return (
    <>
      <HeaderStyle>
        <ImgLogoStyle src="https://i.imgur.com/8aAwol7.png" alt="logo" />{' '}
        <DivEnd>
          <FontAwesomeIcon onClick={() => history.push('/cart')}
          icon={faShoppingCart} />
          <FontAwesomeIcon
            onClick={() => history.push('/login')}
            icon={faUser}
          />
        </DivEnd>
      </HeaderStyle>
    </>
  );
};

export default withRouter(Navbar);
