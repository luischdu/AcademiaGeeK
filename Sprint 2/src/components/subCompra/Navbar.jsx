import React from 'react'
import {FontAwesomeIcon} from '@fortawesome/react-fontawesome'
import {faChevronLeft, faShoppingCart} from '@fortawesome/free-solid-svg-icons'
import styled from 'styled-components'
import { Link } from 'react-router-dom'
import { useHistory } from "react-router-dom";


const NavStyle = styled.div`
    display: flex;
    justify-content: space-between;
    align-items: center;
    padding: 25px 20px 0 20px;

`

export const Navbar = () => {
    let history = useHistory();

    function handleClick() {
      history.push("/");
    }
    return (
        <NavStyle>
            <FontAwesomeIcon onClick={handleClick} icon={faChevronLeft} />
            <Link to='/cart'><FontAwesomeIcon icon={faShoppingCart} /></Link>
        </NavStyle>
    )
}
