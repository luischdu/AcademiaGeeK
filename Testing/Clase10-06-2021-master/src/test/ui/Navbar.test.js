import { mount } from "enzyme";
import React from "react";
import { MemoryRouter } from "react-router";
import { Navbar } from "../../components/ui/Navbar";
import { AuthContext } from "../../auth/AuthContext";
import "@testing-library/jest-dom";
import { types } from "../../types/types";

describe("Validar Navbar", () => {
  const historyMock = {
    push: jest.fn(),
    replece: jest.fn(),
    location: {},
    listen: jest.fn(),
    createHref: jest.fn(),
  };

  const context = {
    dispatch: jest.fn(),
    user: {
      logged: true,
      name: "Carlos",
    },
  };

  const wrapper = mount(
    <AuthContext.Provider value={context}>
      <MemoryRouter>
        <Navbar />
      </MemoryRouter>
    </AuthContext.Provider>
  );
  test("Validar contenido ", () => {
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("span").text().trim()).toBe("Carlos");
  });

  test("Verificar boton ", () => {
    wrapper.find("button").prop("onClick")(); //haciendo clic
    expect(context.dispatch).toHaveBeenCalled({
      type: types.logout,
    }); // verificando que el elemento history se haya dispachado
    expect(historyMock.replece).toHaveBeenCalledWith("/login");
  });
}); 
