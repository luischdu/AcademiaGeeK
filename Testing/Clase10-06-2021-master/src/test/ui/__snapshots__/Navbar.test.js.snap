// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Validar Navbar Validar contenido  1`] = `
<nav
  className="navbar navbar-expand-sm navbar-dark bg-dark"
>
  <a
    className="navbar-brand"
    href="/"
    onClick={[Function]}
  >
    Asociaciones
  </a>
  <div
    className="navbar-collapse"
  >
    <div
      className="navbar-nav"
    >
      <a
        aria-current={null}
        className="nav-item nav-link"
        href="/marvel"
        onClick={[Function]}
      >
        Marvel
      </a>
      <a
        aria-current={null}
        className="nav-item nav-link"
        href="/dc"
        onClick={[Function]}
      >
        DC
      </a>
      <a
        aria-current={null}
        className="nav-item nav-link"
        href="/search"
        onClick={[Function]}
      >
        Search
      </a>
    </div>
  </div>
  <div
    className="navbar-collapse collapse w-100 order-3 dual-collapse2"
  >
    <ul
      className="navbar-nav ml-auto"
    >
      <span
        className="nav-item nav-link text-info"
      >
        Carlos
      </span>
      <button
        className="nav-item nav-link btn"
        onClick={[Function]}
      >
        Logout
      </button>
    </ul>
  </div>
</nav>
`;
